share/functions/__fish_cache_sourced_completions.fish

```diff
@@ -0,0 +1,45 @@
+function __fish_cache_sourced_completions
+    # Allow a `--name=foo` option which ends up in the filename.
+    argparse -s name= -- $argv
+    or return
+
+    set -q argv[1]
+    or return 1
+    
+    set -l cmd (command -s $argv[1])
```

Comment:
maybe also apply `realpath` so the timestamp logic also works if the executable path is a symlink (super rare for the commands in question I guess)

Reply:
We do `stat` for the mtime, not `lstat`, so we should get the real file.

Even if we didn't, in the worst case we'd be running the command one time too often.